app/controllers/general_controller.rb

```diff
@@ -0,0 +1,17 @@
+class GeneralController < ActionController::Base
+  def index
+    @user = current_user
+    @recipe = Recipe.where(user_id: @user.id)
+    @recipe_foods = RecipeFood.all
+    @foods = Food.where(user_id: @user.id)
+    @recipes = @user.recipes
+    @total = @recipes.map { |x| Recipe.total_value(x.id) }.reduce(:+)
+    @food_count = 0
+    @recipes.each do |recipe|
+      recipe_foods = recipe.recipeFoods
+      recipe_foods.each do |_food|
+        @food_count += 1
+      end
+    end
```

Comment:
- [ ] Still have the issue when I click on the `Add food`, `Recipes`, and `General Shopping List` buttons before login in or registering a user I got the error. Please direct the user to the login page when a user clicks on those buttons before creating the account.

For that, you can use `before_action :authenticate_user!` line just after the class. Please add this to all the class where you need to authenticate the user.

Hint:
```ruby
class GeneralController < ActionController::Base
  before_action :authenticate_user!
  
  .....
end
```

Reply:
Okay thanks we'll add that.

Reply:
We've set it up so that if a user is not logged in, the user will be taken directly to the log in page and as a result the user will not be able to click on those buttons

Reply:
Hi @Kingstalux,

It's not only about clicking the button, suppose if we go directly by writing the URL `http://127.0.0.1:3000/general` then also it should redirect me to the log-in page otherwise, the user can easily go to all your routes and this leads to security issues in your application.

Hope you get a clear idea.
Thank You

Reply:
Ahh okay, I get it now. Thanks for clarifying this.